search-app/src/cmr/search/services/humanizers/humanizer_report_service.clj

```diff
@@ -70,18 +75,18 @@
   (map #(rfm->umm-collection context %) rfms))
 
 (defn- wait-and-retry-for-collection-cache
-  "Wait configurable number of seconds before retrying to get all the collections
-   from collection cache. If collections are still not present, throw warning."
-  [context]
-  (info (format "Collection cache is not populated after %d seconds of delay" 
-                (humanizer-report-generator-job-delay)))
-  (Thread/sleep (humanizer-report-generator-job-wait)) 
-  (let [rfms (metadata-cache/all-cached-revision-format-maps context)]
-    (if (seq rfms)
-      rfms
-      (warn (format "Collection cache is not populated after %d seconds of delay and %d milli-seconds of wait."
-                    (humanizer-report-generator-job-delay)
-                    (humanizer-report-generator-job-wait))))))
+  "Wait configurable number of milli seconds before retrying configurable number of times
+   to get all the collections from collection cache."
+  [count context]
+  (when (< count (retry-count))
+    (info (format (str "Humanizer report generator job is sleeping for %d milli-seconds"
+                       "before retrying to see if the collection cache is populated.")
+                  (humanizer-report-generator-job-wait)))
+    (Thread/sleep (humanizer-report-generator-job-wait))
+    (let [rfms (metadata-cache/all-cached-revision-format-maps context)]
+      (if (seq rfms)
+        rfms
+        (recur (inc count) context)))))
 
 (defn- get-all-collections
   "Retrieves all collections from the Metadata cache, partitions them into batches of size
```

Comment:
I think you can refactor this function to something like:
```
(if-let [rfms (get-cached-revision-format-maps context)]
 (map
     #(rfms->umm-collections context %)
     (partition-all (humanizer-report-collection-batch-size) rfms))
(warn (format "Collection cache is not populated after %d seconds of delay and %d times of retry." ...)))
```
And in function `get-cached-revision-format-maps`, you have a loop that tries to call ` (metadata-cache/all-cached-revision-format-maps context)` returns it if it is not empty; otherwise retries until the retry count is exhausted, i.e. something like:
```
(loop [retries (retry-count)]
  (if-let [rfms (metadata-cache/all-cached-revision-format-maps context)] 
    rfms
   (when (> retries 0)
     ;; logging and sleeping, then 
     (recur (dec max-tries))))
```
Add a `(seq ...)` call in `all-cached-revision-format-maps` function, so the return value will be nil when empty.